dss/logging.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from logging import NOTSET, DEBUG, INFO, WARNING, ERROR, CRITICAL
+import sys
+
+from typing import Mapping, Union, Sequence, Tuple
+
+import dss
+from dss.config import Config
+
+log_level_t = Mapping[Union[None, str, logging.Logger], Tuple[int, ...]]
+"""
+Values of this type map a given logger and debug level to the log level for that logger at the given debug level. The
+debug level is an application specific concept (:py:meth:`dss.Config.debug_level`). If the debug level exceeds the
+length of the tuple, the last tuple element is used.
+"""
+
+main_log_levels: log_level_t = {
+    None: (WARNING, INFO, DEBUG),
+    dss.logger: (INFO, DEBUG),
+    'botocore.vendored.requests.packages.urllib3.connectionpool': (WARNING, WARNING, DEBUG)
+}
+"""
+The main log level map.
+"""
+
+test_log_levels: log_level_t = {
+    dss.logger: (WARNING, DEBUG),
+}
+"""
+The log levels for running tests. The test map overrides and extends the main map.
+"""
+
+
+def configure_cli_logging():
+    """
+    Prepare logging for use in a command line application.
+    """
+    _configure_logging(stream=sys.stderr)
+
+
+def configure_daemon_logging():
+    """
+    Prepare logging for use within a AWS Lambda function.
+    """
+    _configure_logging(stream=sys.stdout,
+                       # Timestamp is handled by AWS Lambda
+                       format='%(name)s %(levelname)s %(message)s')
+
+
+def configure_test_logging():
+    """
+    Configure logging for use during unit tests.
+    """
+    _configure_logging(stream=sys.stderr, test=True)
+
+
+def _configure_logging(test=False, **kwargs):
+    assert len(logging.getLogger().handlers) == 0, "Logging was already configured"
+    logging.basicConfig(**kwargs)
+    debug = Config.debug_level()
+    log_levels = main_log_levels
+    if test:
+        log_levels = log_levels.copy()
+        log_levels.update(test_log_levels)
+    for logger, levels in log_levels.items():
+        if isinstance(logger, (str, type(None))):
+            logger = logging.getLogger(logger)
+        level = levels[min(debug, len(levels) - 1)]
+        logger.setLevel(level)
+
+
+def defuse_flask_logging(flask_app):
```

Comment:
Could it be easier to rename the Flask app? Does that avoid the need for this method?

Reply:
Let me try that out.

Reply:
There was an even easier way:

https://github.com/HumanCellAtlas/data-store/pull/859/commits/c120f63492132b95c732314051a5df134bcfc720#diff-83c83ff7781521cdb005c5d2180b2522L187